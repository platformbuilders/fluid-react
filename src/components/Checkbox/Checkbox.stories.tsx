import { expect, jest } from '@storybook/jest';
import type { Meta, StoryObj } from '@storybook/react';
import { configure, within } from '@storybook/testing-library';
import Checkbox from './index';
configure({
  testIdAttribute: 'id',
});

//  Mocks
const eventMockFn = jest.fn();

const testId = 'checkbox-id';

const meta: Meta<typeof Checkbox> = {
  title: 'Components/Checkbox',
  component: Checkbox,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  args: {
    onChange: eventMockFn,
    id: testId,
    label: 'Enable Settings',
  },
  play: async ({ canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('Checkbox - Normal | Test Render', async () => {
      expect(canvas.getByTestId(testId)).toBeInTheDocument();
      expect(canvas.getByTestId(testId)).not.toBeChecked();
    });
  },
};

export default meta;

type Story = StoryObj<typeof Checkbox>;

export const Normal: Story = {};
