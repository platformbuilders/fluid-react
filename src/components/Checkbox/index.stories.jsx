import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';
import markdown from './checkbox.md';
import CheckBox from './index';

storiesOf('CheckBox', module)
  .addDecorator(withKnobs)
  .addParameters({
    readme: {
      content: markdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => (
    <CheckBox
      checked={boolean('checked', false)}
      label={text('label', 'Enable Notification')}
      error={text('error', 'here is error message')}
      name={text('name', 'message')}
    />
  ));
