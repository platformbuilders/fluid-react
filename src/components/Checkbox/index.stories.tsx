import type { Meta, StoryObj } from '@storybook/react';
import CheckBox from './index';

const meta: Meta<typeof CheckBox> = {
  title: 'CheckBox',
  component: CheckBox,
  args: {
    checked: false,
    label: 'Enable Notification',
    error: 'here is error message',
    name: 'message',
  },
};

export default meta;
type Story = StoryObj<typeof CheckBox>;

export const Default: Story = {};
