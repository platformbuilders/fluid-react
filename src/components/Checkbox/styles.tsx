import styled from 'styled-components';
import { getTheme } from '@platformbuilders/theme-toolkit';

const brandPrimaryMain = getTheme('brand.primary.main');
const brandPrimaryContrast = getTheme('brand.primary.contrast');

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  margin-bottom: 10px;
`;

export const CheckboxRoot = styled.label`
  z-index: 0;
  position: relative;
  display: inline-block;
  color: rgba(0, 0, 0, 0.87);
  font-size: 16px;
  line-height: 1.5;

  input:checked + span::before,
  input:indeterminate + span::before {
    border-color: ${brandPrimaryMain};
    background-color: ${brandPrimaryMain};
  }

  input:checked + span::after,
  input:indeterminate + span::after {
    border-color: ${brandPrimaryContrast};
  }

  input:indeterminate + span::after {
    border-left: none;
    transform: translate(4px, 3px);
  }

  &:hover {
    input {
      opacity: 0.12;

      :focus {
        opacity: 0.16;
      }
    }
  }

  input:active + span::before {
    border-color: ${brandPrimaryMain};
  }

  input:checked:active + span::before {
    border-color: transparent;
    background-color: rgba(0, 0, 0, 0.6);
  }

  input:disabled + span {
    color: rgba(0, 0, 0, 0.38);
    cursor: initial;
  }

  input:disabled + span::before {
    border-color: currentColor;
  }

  input:checked:disabled + span::before,
  input:indeterminate:disabled + span::before {
    border-color: transparent;
    background-color: currentColor;
  }
`;

export const Check = styled.input`
  appearance: none;
  z-index: -1;
  position: absolute;
  left: -10px;
  top: -8px;
  display: block;
  margin: 0;
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: rgba(0, 0, 0, 0.6);
  box-shadow: none;
  outline: none;
  opacity: 0;
  transform: scale(1);
  pointer-events: none;
  transition: opacity 0.3s, transform 0.2s;

  &:checked,
  &:indeterminate {
    background-color: ${brandPrimaryMain};
  }

  &:focus {
    opacity: 0.04;
  }

  &:active {
    opacity: 1;
    transform: scale(0);
    transition: transform 0s, opacity 0s;
  }

  &:disabled {
    opacity: 0;
  }
`;

export const Label = styled.span`
  display: inline-block;
  width: 100%;
  cursor: pointer;

  &::before {
    content: '';
    display: inline-block;
    box-sizing: border-box;
    margin: 3px 11px 3px 1px;
    border: solid 2px; /* Safari */
    border-color: rgba(0, 0, 0, 0.6);
    border-radius: 2px;
    width: 18px;
    height: 18px;
    vertical-align: top;
    transition: border-color 0.2s, background-color 0.2s;
  }

  &::after {
    content: '';
    display: block;
    position: absolute;
    top: 3px;
    left: 1px;
    width: 10px;
    height: 5px;
    border: solid 2px transparent;
    border-right: none;
    border-top: none;
    transform: translate(3px, 4px) rotate(-45deg);
  }
`;
