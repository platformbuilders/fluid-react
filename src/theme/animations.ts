import LoadingAnimation from '../assets/animations/loading.json';
import { Animations } from '../types';

export default {
  circularLoading: LoadingAnimation,
  buttonLoading: LoadingAnimation,
  linearLoading: LoadingAnimation,
} as Animations;
