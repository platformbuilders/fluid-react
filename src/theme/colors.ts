import { lighten, darken } from 'polished';
import { Colors } from '../utils/types';

const primary = '#C6B09E';
const secondary = '#805281';
const tertiary = '#2A2E39';
const accent = '#547E6A';

const colors: Colors = {
  failure: '#cc0000',
  warning: '#F5B800',
  success: '#357a38',
  transparent: 'transparent',
  disabled: '#EBEBE4',
  primary: {
    light: lighten(0.05, primary),
    main: primary,
    dark: darken(0.12, primary),
    contrast: '#ffffff',
  },
  secondary: {
    light: lighten(0.05, secondary),
    main: secondary,
    dark: darken(0.1, secondary),
    contrast: '#ffffff',
  },
  tertiary: {
    light: lighten(0.15, tertiary),
    main: tertiary,
    dark: darken(0.1, tertiary),
    contrast: '#ffffff',
  },
  accent: {
    light: lighten(0.05, accent),
    main: accent,
    dark: darken(0.05, accent),
    contrast: '#ffffff',
  },
};

export default colors;
