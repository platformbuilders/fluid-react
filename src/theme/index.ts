export { default as colors } from './colors';
export { default as radius } from './radius';
export { default as animations } from './animations';
