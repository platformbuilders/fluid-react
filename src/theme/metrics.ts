import { Dimensions } from 'react-native';
import { verticalScale, moderateScale } from 'react-native-size-matters';
import { Metrics } from '../types';

const { width: screenWidth, height: screenHeight } = Dimensions.get('window');

export default {
  headerSpacing: `${verticalScale(20)}px`,
  sceneSpacing: `${moderateScale(32)}px`,
  smallSpacing: `${moderateScale(8)}px`,
  mediumSpacing: `${moderateScale(12)}px`,
  largeSpacing: `${moderateScale(20)}px`,
  giantSpacing: `${moderateScale(26)}px`,
  sectionSpacing: `${moderateScale(38)}px`,
  statusHeight: `${moderateScale(44)}px`,
  smallRadius: `${moderateScale(3)}px`,
  mediumRadius: `${moderateScale(8)}px`,
  largeRadius: `${moderateScale(15)}px`,
  modalRadius: `${moderateScale(20)}px`,
  buttonRadius: `${moderateScale(38)}px`,
  screenWidth,
  screenHeight,
} as Metrics;
