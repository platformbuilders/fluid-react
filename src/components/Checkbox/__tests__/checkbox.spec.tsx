import { ThemeProvider } from 'styled-components';
import { fireEvent, render } from '@testing-library/react';

import Checkbox, { CheckboxProps } from '..';
import theme from '../../../theme';

const defaultProps: CheckboxProps = {
  checked: false,
  onChange: jest.fn(),
  label: 'Text',
  name: 'Text',
};

describe('Component: Checkbox', () => {
  test('snapshots with default props', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Checkbox {...defaultProps} />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  test('should call onPress when pressed', () => {
    // should
    const mockFunction = jest.fn();
    const { getByRole } = render(
      <Checkbox {...defaultProps} onChange={mockFunction} />,
    );

    expect(mockFunction).not.toHaveBeenCalled();

    fireEvent.click(getByRole('checkbox'));
    expect(mockFunction).toHaveBeenCalled();
  });
});
