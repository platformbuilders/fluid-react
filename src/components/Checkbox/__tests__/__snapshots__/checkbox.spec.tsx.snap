// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Checkbox snapshots with default props 1`] = `
<div>
  <div
    class="sc-dkzDqf hSgqxi"
  >
    <label
      class="MuiFormControlLabel-root"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root sc-hKMtZM dQXgIx sc-eCYdqJ fluid-checkbox"
        label="Text"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-4"
            data-indeterminate="false"
            name="Text"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Text
      </span>
    </label>
  </div>
</div>
`;
